plugins/file/plugin.go

```diff
@@ -113,10 +176,33 @@ func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
 
 // Handler4 handles DHCPv4 packets for the file plugin
 func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
-	// TODO check the MAC address in the request
-	//      if it is present in StaticRecords, forge a response
-	//      and stop processing.
-	return resp, true
+	record, ok := DHCPv4Records[req.ClientHWAddr.String()]
+	if !ok {
+		log.Printf("plugins/file: MAC address %s is new, leasing new IP address", req.ClientHWAddr.String())
+		record, ok = createIP(serverIP, netmask)
+		if !ok {
+			log.Printf("plugins/file: no new IP addresses available")
+			return resp, true
+
+		}
+		err := saveIPAddress(record, req.ClientHWAddr)
+		if err != nil {
+			log.Printf("plugins/file: SaveIPAddress failed: %v", err)
+		}
+		DHCPv4Records[req.ClientHWAddr.String()] = record
+
+	}
+	ipaddr := record.IP
+	log.Printf("plugins/file: found IP address %s for MAC %s", ipaddr, req.ClientHWAddr.String())
+	if req == nil {
+		log.Printf("plugins/file: Packet is nil!")
+	}
+	reply, err := dhcpv4.NewReplyFromRequest(req, dhcpv4.WithYourIP(ipaddr), dhcpv4.WithLeaseTime(LeaseTime))
```

Comment:
You shouldn't create a `NewReplyFromRequest` here. It is already created in the `MainHandler`: https://github.com/coredhcp/coredhcp/blob/master/coredhcp.go#L175